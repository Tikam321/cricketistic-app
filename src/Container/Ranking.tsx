import { useEffect, useRef, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import '../assets/css/ranking.css';
import { rankingFormat, tempRankinglIST } from '../helper/rankingHelper';
import { getOdiRankinglist, getT20RankingList, getTestRankinglist } from "../store/actions/rankingActions";
import { RootState } from "../store/reducer/rootReducer";

export interface rankingObjectType {
    countryName: string,
    rankingPoints: number,
    id: number,
    position: number
}

const rankingData: rankingObjectType[] = [{  
        countryName: "India",
        rankingPoints: 1,
        id: 1,
        position: 1
}];

const Ranking = () => {

    const dispatcher =  useDispatch();
    const { t20RankingList, odiRankingList, testRankingList } = useSelector((rootState: RootState) => rootState.ranking);
    const [t20Rankings, setT20Rankings] = useState<rankingObjectType[]>([]);
    const [odiRankings, setOdiRankings] = useState<rankingObjectType[]>([]);
    const [testRankings, setTestRankings] = useState<rankingObjectType[]>([]);
    const tooltipRef = useRef<HTMLSpanElement>(null);

    const [format, setFormat] = useState<string>("TEST");

    useEffect(() => {
        console.log("dispatch action is called");
        dispatcher(getT20RankingList());
        dispatcher(getOdiRankinglist());
        dispatcher(getTestRankinglist());
    }, []);

    useEffect(() => {
        if (typeof t20RankingList !== "undefined") {
            setT20Rankings(t20RankingList);
            console.log(t20RankingList);
        } else {
            console.log("the ranklist is undefined");
            
        }
    }, [t20RankingList]);

    useEffect(() => {
        if (typeof odiRankingList !== "undefined") {
            setOdiRankings(odiRankingList);
            console.log(t20RankingList);
        } else {
            console.log("the ranklist is undefined");
            
        }
    }, [odiRankingList]);

    useEffect(() => {
        if (typeof testRankingList !== "undefined") {
            setTestRankings(testRankingList);
            console.log(t20RankingList);
        } else {
            console.log("the ranklist is undefined");
            
        }
    }, [testRankingList]);

    const getRankings = (format: string) => {
        if (format === rankingFormat.T20) {
            return t20Rankings;
        } else if (format === rankingFormat.ODI) {
            return odiRankings;
        } else {
            return testRankings;
        }
    };
    useEffect(() => {
        const tooltipSpan: any = document.querySelectorAll(".tooltip");
        console.warn(tooltipSpan);
        window.onmousemove = function (event) {
            var x1 = (event.clientX - 100) + "px";
            var y1 = (event.clientY - 30) + "px";
            console.log(x1);
            
            for (var i = 0; i < tooltipSpan.length; i++) {
                tooltipSpan[i].style.top = y1;
                tooltipSpan[i].style.left = x1;
            }
        }
    }, []);
    return (
        <>

            <div className="main-ranking ranking-container container">
         {/* <CSVLink data={rankingList} > Download ranking</CSVLink>   */}
         {/* <DownloadTableExcel
                    filename="users table"
                    sheet="users"
                    currentTableRef={tableRef.current}
                >

                   <button> Export excel </button>

                </DownloadTableExcel> */}
                {/* <button onClick={event => ExportToExcel(rankingList, "myFile")}> Export excel </button> */}
                <h2>ICC Cricket Ranking - Men's Batting </h2>
                <div className="ranking-btn"> 
                <button onClick={() => setFormat("TEST")} className={`${format === "TEST" ? "activated" : ""}`}> TEST </button>
                <button onClick={() => setFormat("ODI")}className={`${format === "ODI" ? "activated" : ""}`}> ODI</button>
                <button onClick={() => setFormat("T20")} className={`${format === "T20" ? "activated" : ""}`}> T20</button>
                 </div>
                <div className="card">
                <table>
                 <tbody>
                    <tr>
                        <th>Ranking</th>
                        <th>Country Name</th>
                        <th>Position</th>
                        <th>Rating</th>
                    </tr>
                    { tempRankinglIST.map((item: rankingObjectType) => (
                        <tr>
                            <td className="relative-tooltip" >{item.position}
                            <span ref={tooltipRef} className="tooltip">{item.position}</span> 
                            </td>
                            <td className="relative-tooltip" >{item.countryName}
                            <span className="tooltip">{item.countryName}</span> 
                            </td>
                            <td className="relative-tooltip" >{item.position}
                            <span className="tooltip">{item.position}</span> 
                            </td>
                            <td className="relative-tooltip" > {item.rankingPoints}
                            <span className="tooltip">{item.rankingPoints}</span> 
                            </td>
                        </tr>
                    ))}
                  </tbody>
                </table>
                </div>
</div>
        </>
    );
}
export default Ranking;