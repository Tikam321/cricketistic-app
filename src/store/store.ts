import { EnhancedStore, configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
// import createRootReducer from "./reducer/rootReducer";
import {createLogger} from "redux-logger";
import { createRootReducer } from "./reducer/rootReducer";
import createSagaMiddleware from "@redux-saga/core";
import { getT20Ranking, rankingSaga } from "../saga/rankingSaga";
import { authSaga } from "../saga/authSaga";
import rootSaga from "../saga/rootSaga";
// import createRootReducer  from "./reducer/rootReducer";

const middleware = getDefaultMiddleware();
const sagaMiddleware = createSagaMiddleware();
// middleWare.push(logger)
const logger = createLogger({
    level: "info",
    collapsed: true
});

//  type  store1 = EnhancedStore;

middleware.push(logger);
middleware.push(sagaMiddleware);
export const store: EnhancedStore = configureStore({
    reducer: createRootReducer,
    middleware
  });
sagaMiddleware.run(rootSaga);
// export default store;
