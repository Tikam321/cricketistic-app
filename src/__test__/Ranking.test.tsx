import { createEvent, fireEvent, getByText, render, screen } from '@testing-library/react';
import Ranking from '../Container/Ranking';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../store/store';
import configureStore from 'redux-mock-store';
import { Component } from 'react';

const withoutData  = {
  t20RankingList: [],
  odiRankingList: [],
  testRankingList: []
}

const withData = {
  t20RankingList: [{
    countryName: "India",
    id: 1,
    position: 1,
    rankingPoints: 1
}],
odiRankingList: [{
    countryName: "India",
    id: 1,
    position: 1,
    rankingPoints: 1
}],
testRankingList: [{
    countryName: "India",
    id: 1,
    position: 1,
    rankingPoints: 1
}],
}

// const rankingReducer = {
//   t20RankingList: [],
//   odiRankingList: [],
//   testRankingList: []
// }

const mockStore = configureStore([]);

const renderwithData = () => {
  const store = mockStore({
    ranking: withData
   });
    return render( <Provider store={store}>
     <BrowserRouter>
     <Ranking />
     </BrowserRouter>
   </Provider>);
}

test("match snapshot without data DOM", () => {
  const tree = renderwithData();
  expect(tree).toMatchSnapshot();
});

test ("render the ranking page  without data", async () => {
  const store = mockStore({
    ranking: {}
   });
   
    render( <Provider store={store}>
     <BrowserRouter>
     <Ranking />
     </BrowserRouter>
   </Provider>);
});

test ("render the ranking page  with data", async () => {
  const store = mockStore({
    ranking: withData
   });
   
    render( <Provider store={store}>
     <BrowserRouter>
     <Ranking />
     </BrowserRouter>
   </Provider>);
});

test('fireing click event for different format ranking ', async () => {
  //given
  const store1 = mockStore({
   ranking: withData
  });
  
   render( <Provider store={store1}>
    <BrowserRouter>
    <Ranking />
    </BrowserRouter>
  </Provider>);

   expect(screen.getByRole('title')).toHaveTextContent("CC Cricket Ranking - Men's Batting");
   fireEvent.click(screen.getByTestId("odiButton"));
   fireEvent.click(screen.getByTestId("testButton"));
   fireEvent.click(screen.getByTestId("t20Button"));
});

test("onmouseover for tabledata", async () => {
  const component =  renderwithData();
  const createEvent1 =  createEvent.mouseMove(component.getByTestId('tableData'), {
    clientX: 595,
    clientY: 535,
  })
  fireEvent(component.getByTestId('tableData'), createEvent1);
  component.debug();
});