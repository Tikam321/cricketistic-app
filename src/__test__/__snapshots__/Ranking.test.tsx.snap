// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot without data DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-ranking ranking-container container"
      >
        <h2
          role="title"
        >
          ICC Cricket Ranking - Men's Batting
        </h2>
        <div
          class="ranking-btn"
        >
          <button
            class="activated"
            data-testid="testButton"
          >
             TEST 
          </button>
          <button
            class=""
            data-testid="odiButton"
          >
             ODI
          </button>
          <button
            class=""
            data-testid="t20Button"
          >
             T20
          </button>
        </div>
        <div
          class="card"
        >
          <table>
            <tbody>
              <tr>
                <th>
                  Ranking
                </th>
                <th>
                  Country Name
                </th>
                <th>
                  Position
                </th>
                <th>
                  Rating
                </th>
              </tr>
              <tr>
                <td
                  class="relative-tooltip"
                >
                  1
                  <span
                    class="tooltip"
                  >
                    1
                  </span>
                </td>
                <td
                  class="relative-tooltip"
                  data-testid="tableData"
                >
                  India
                  <span
                    class="tooltip"
                  >
                    India
                  </span>
                </td>
                <td
                  class="relative-tooltip"
                >
                  1
                  <span
                    class="tooltip"
                  >
                    1
                  </span>
                </td>
                <td
                  class="relative-tooltip"
                >
                   
                  1
                  <span
                    class="tooltip"
                  >
                    1
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-ranking ranking-container container"
    >
      <h2
        role="title"
      >
        ICC Cricket Ranking - Men's Batting
      </h2>
      <div
        class="ranking-btn"
      >
        <button
          class="activated"
          data-testid="testButton"
        >
           TEST 
        </button>
        <button
          class=""
          data-testid="odiButton"
        >
           ODI
        </button>
        <button
          class=""
          data-testid="t20Button"
        >
           T20
        </button>
      </div>
      <div
        class="card"
      >
        <table>
          <tbody>
            <tr>
              <th>
                Ranking
              </th>
              <th>
                Country Name
              </th>
              <th>
                Position
              </th>
              <th>
                Rating
              </th>
            </tr>
            <tr>
              <td
                class="relative-tooltip"
              >
                1
                <span
                  class="tooltip"
                >
                  1
                </span>
              </td>
              <td
                class="relative-tooltip"
                data-testid="tableData"
              >
                India
                <span
                  class="tooltip"
                >
                  India
                </span>
              </td>
              <td
                class="relative-tooltip"
              >
                1
                <span
                  class="tooltip"
                >
                  1
                </span>
              </td>
              <td
                class="relative-tooltip"
              >
                 
                1
                <span
                  class="tooltip"
                >
                  1
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
